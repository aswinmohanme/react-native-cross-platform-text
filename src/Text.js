import React from 'react';
import { Text as ReactNativeText } from 'react-native';
import FontStyleWrapper from './FontStyleWrapper';

const Text = ({ style, ...props }) => (
  <FontStyleWrapper style={style}>
    {fontStyle => <ReactNativeText style={fontStyle} {...props} />}
  </FontStyleWrapper>
);

export default Text;
