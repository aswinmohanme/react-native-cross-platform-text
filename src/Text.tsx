import * as React from "react";
import { Text as ReactNativeText, StyleProp, TextStyle } from "react-native";
import FontStyleWrapper from "./FontStyleWrapper";

interface TextProps {
  // https://basarat.gitbooks.io/typescript/docs/types/index-signatures.html
  [k: string]: any;
  style: StyleProp<TextStyle>;
}

const Text = ({ style, ...props }: TextProps) => (
  <FontStyleWrapper style={style}>
    {(fontStyle) => <ReactNativeText style={fontStyle} {...props} />}
  </FontStyleWrapper>
);

export default Text;
