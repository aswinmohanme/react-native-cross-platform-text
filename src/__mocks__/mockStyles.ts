import { StyleSheet } from 'react-native';

const doMockStyles = StyleSheet.create({
  fontStyle: {
    fontWeight: 'normal',
    fontFamily: 'Arial',
  },
  section: {
    paddingHorizontal: 10,
  },
  container: {
    margin: 20,
  },
});

export default doMockStyles;
