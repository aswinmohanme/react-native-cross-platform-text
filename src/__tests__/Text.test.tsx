import * as React from "react";
import { Platform } from "react-native";
import * as ShallowRenderer from "react-test-renderer/shallow";
import Text from "../Text";
import mockStyles from './__mocks__/mockStyles';

jest.mock("Platform");

it("iOS returns fontFamily and weight for normal", () => {
  Platform.OS = "ios";
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(<Text style={mockStyles.fontStyle}>this is iOS</Text>);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it("Android returns fontFamily-Regular and weight = undefined for normal", () => {
  Platform.OS = "android";
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(<Text style={mockStyles.fontStyle}>this is Android</Text>);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it("flattens styles", () => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(
    <Text
      style={[
        mockStyles.section,
        [mockStyles.container, mockStyles.fontStyle],
      ]}
    >works with nested styles!</Text>
  );
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
