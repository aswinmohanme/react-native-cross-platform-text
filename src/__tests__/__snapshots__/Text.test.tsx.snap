// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android returns fontFamily-Regular and weight = undefined for normal 1`] = `
<FontStyleWrapper
  style={
    Object {
      "fontFamily": "Arial",
      "fontWeight": "normal",
    }
  }
>
  [Function]
</FontStyleWrapper>
`;

exports[`flattens styles 1`] = `
<FontStyleWrapper
  style={
    Array [
      Object {
        "paddingHorizontal": 10,
      },
      Array [
        Object {
          "margin": 20,
        },
        Object {
          "fontFamily": "Arial",
          "fontWeight": "normal",
        },
      ],
    ]
  }
>
  [Function]
</FontStyleWrapper>
`;

exports[`iOS returns fontFamily and weight for normal 1`] = `
<FontStyleWrapper
  style={
    Object {
      "fontFamily": "Arial",
      "fontWeight": "normal",
    }
  }
>
  [Function]
</FontStyleWrapper>
`;

exports[`passes Text props through 1`] = `
<FontStyleWrapper
  style={
    Object {
      "fontFamily": "Arial",
      "fontWeight": "normal",
    }
  }
>
  [Function]
</FontStyleWrapper>
`;
